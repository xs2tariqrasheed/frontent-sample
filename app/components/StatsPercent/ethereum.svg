<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16">
    <path fill="#655BF5" d="M11.333 8L8 10 4.667 8 8 2l3.333 6zM8 11.333l-3.333-2L8 14l3.333-4.667-3.333 2z"/>
</svg>
