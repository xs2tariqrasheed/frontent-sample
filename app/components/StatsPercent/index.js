import React from 'react';
import PropTypes from 'prop-types';
import {
  TotalLine,
  CurrentSkill,
  PotentialSkill,
  SkillsWrapper,
} from './styled';

const StatsPercent = ({ percent, potentialPercent }) => (
  <SkillsWrapper>
    <CurrentSkill width={percent} />
    <PotentialSkill width={potentialPercent} />
    <TotalLine width={100 - (percent + potentialPercent)} />
  </SkillsWrapper>
);

StatsPercent.propTypes = {
  percent: PropTypes.number,
  potentialPercent: PropTypes.number,
};

export default StatsPercent;
