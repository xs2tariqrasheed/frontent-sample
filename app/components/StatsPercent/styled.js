import styled from 'styled-components';

export const SkillsWrapper = styled.div`
  position: relative;
  width: 100%;
`;

export const CurrentSkill = styled.div`
  position: absolute;
  left: 0;
  top: 0;
  z-index: 2;
  width: ${props => props.width}%;
  height: 4px;
  background-color: #ffc530;
`;

export const PotentialSkill = styled.div`
  position: absolute;
  left: 0;
  top: 0;
  z-index: 1;
  width: ${props => props.width}%;
  height: 4px;
  background-color: #655bf5;
`;

export const TotalLine = styled.div`
  position: absolute;
  left: 0;
  top: 0;
  z-index: 0;
  height: 4px;
  width: ${props => props.width}%;
  width: 100%;
  background-color: #d8d6fc;
`;
