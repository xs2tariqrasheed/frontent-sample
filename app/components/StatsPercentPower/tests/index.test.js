import React from 'react';
import { render } from 'react-testing-library';

import StatsPercent from '../index';

describe('<StatsPercent percent={45} />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<StatsPercent percent={45} potentialPercent={100} />);
    expect(firstChild).toMatchSnapshot();
  });
});
