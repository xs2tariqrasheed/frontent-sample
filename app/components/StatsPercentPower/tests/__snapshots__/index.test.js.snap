// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatsPercent percent={45} /> should render and match the snapshot 1`] = `
<div
  class="styled__SkillsWrapper-sc-1syepi9-0 gIGjdo"
>
  <div
    class="styled__CurrentSkill-sc-1syepi9-1 cELXRB"
    width="45"
  />
  <div
    class="styled__PotentialSkill-sc-1syepi9-2 gGqLVy"
    width="100"
  />
  <div
    class="styled__TotalLine-sc-1syepi9-3 gNXImb"
    width="-45"
  />
</div>
`;
